torchtune/datasets/_chat.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Callable, Dict, Generator, List, Tuple
+
+from datasets import load_dataset
+from torch.utils.data import Dataset
+from torchtune.config._utils import _get_template
+
+from torchtune.data import (
+    Dialogue,
+    PromptTemplate,
+    Sample,
+    tokenize_prompt_and_response,
+    truncate_if_necessary,
+)
+from torchtune.modules import Tokenizer
+
+
+class ChatDataset(Dataset):
+    """
+    Class that supports any custom dataset with multiturn conversations.
```

Comment:
Is this true? This only supports custom datasets from the datasets repo?

Reply:
I mean that users can use this to configure any local or huggingface dataset, just like `InstructDataset`